xarray/core/dataarray.py

```diff
@@ -244,6 +240,16 @@ def _replace_maybe_drop_dims(self, variable, name=__default):
                                  if set(v.dims) <= allowed_dims)
         return self._replace(variable, coords, name)
 
+    def _replace_indexes(self, indexes):
+        obj = self
+        for dim, idx in iteritems(indexes):
+            if idx.name is None:
+                idx.name = dim + "_unnamed_level"
```

Comment:
Maybe should handle this instead by requiring names for all levels on multi-indexes used by xarray? Or we could pick defaults (e.g., `level_0`, `level_1`, etc.). I worry that this choice is likely to give you multiple levels with the same name "unnamed_level", though I suppose this isn't so bad given that the unnamed level stuff is mostly there as a soft reminder to add names.